std/process.d

```diff
@@ -2815,6 +2815,10 @@ void kill(Pid pid, int codeOrSignal)
     else version (Posix)
     {
         import core.sys.posix.signal : kill;
+        if (pid.osHandle == Pid.invalid)
+            throw new ProcessException("Pid is invalid");
+        if (pid.osHandle == Pid.terminated)
+            throw new ProcessException("Pid is already terminated");
```

Comment:
I suggest being even more defensive here and rejecting all negative values as invalid. I'm not sure how anyone would be able to construct a `Pid` with `osHandle` equallying -3 or lower, but in the event that this ever happens, the code will not do what is expected.

```suggestion
        if (pid.osHandle < 0)
        {
            if (pid.osHandle == Pid.terminated)
                throw new ProcessException("Pid is already terminated");
            throw new ProcessException("Pid is invalid");
        }
```
This also saves a branch in the common case.

Reply:
I pushed 99062799debecc4c2d5f51500e5c7cfd4dbf97b4 but this got merged first.

Reply:
This poses another semantic change. You additionally reject the value 0 as invalid. For POSIX kill, it has the semantic of killing the current process group. I hope this is intended. Maybe the Pid constructor should actually validate the passed id value (or assert the expected invariant)?

Reply:
The `Pid` constructor is `private`, so I don't think it matters that much where this validation is.